services/caldar-api/src/controllers/customer-controller.ts

```diff
@@ -0,0 +1,159 @@
+);
+
+type queryType = {
+  type?: CustomerType;
+};
+
+export const getCustomers: RequestHandler = (req: Request, res: Response) => {
+  let response = getDataCustomers();
+
+  res.status(200).json({
+    status: "Ok",
+    content: response,
+  });
+};
+
+export const getCustomersByType: RequestHandler = (
+  req: Request,
+  res: Response
+) => {
+  const query: queryType = {};
+
+  query.type = req.query.type ? Number(req.query.type) : undefined;
+
+  let customers = getDataCustomers();
+
+  let list = !query.type
+    ? customers
+    : customers.filter((x) => x.type === query.type);
+
+  res.status(200).json({
+    status: "Ok",
+    content: list,
+  });
+};
+
+export const createCustomer: RequestHandler = (req: Request, res: Response) => {
+  const newCustomer: Customer = req.body;
+
+  if (!newCustomer) {
+    res.status(400).json({
+      status: "error",
+      error: "New customer data must be present in the request body.",
+    });
+  }
+
+  let customers = getDataCustomers();
+
+  newCustomer.id = customers[customers.length - 1].id + 1;
+
+  const newList = [...customers, newCustomer];
+
+  saveData(newList);
+
+  res.status(200).json({
+    status: "ok",
+    item: newCustomer,
+  });
+};
+
+export const updateCustomer: RequestHandler = (req: Request, res: Response) => {
+  const updatedCustomer: Customer = req.body;
+
+  if (!updatedCustomer) {
+    res.status(400).json({
+      status: "error",
+      error: "Updated customer data must be present in the request body.",
+    });
+  }
+
+  if (!updatedCustomer.id) {
+    res.status(400).json({
+      status: "error",
+      error:
+        "The ID of the customer to update must be present in the request body.",
+    });
+  }
+
+  let customers = getDataCustomers();
+
+  let existCustomer = customers.some((x) => x.id === updatedCustomer.id);
+
+  if (!existCustomer) {
+    res.status(404).json({
+      status: "not found",
+      error: "Customer not found",
+    });
+  }
+
+  const cleanedList = customers.filter((x) => x.id !== updatedCustomer.id);
+
+  const updatedList: Customer[] = [...cleanedList, updatedCustomer];
+
+  saveData(updatedList);
+
+  res.status(200).json({
+    status: "ok",
+    item: updatedCustomer,
+  });
+};
+
+export const deleteCustomer: RequestHandler = (req, res) => {
+  let customerId = req.query.ID;
+
+  if (!customerId) {
+    res.status(400).json({
+      status: "error",
+      error: "The ID of the customer to delete must be present.",
+    });
+  }
+
+  let customers = getDataCustomers();
+
+  const deletedCustomer = customers.filter(
+    (x) => x.id === Number(customerId)
+  )[0];
+
+  if (!deletedCustomer) {
+    res.status(404).json({
+      status: "not found",
+      error: "Customer not found",
+    });
+  }
+
+  const updatedList = customers.filter((x) => x.id !== Number(customerId));
+
+  saveData(updatedList);
+
+  res.status(200).json({
+    status: "ok",
+    content: deletedCustomer,
+  });
+};
+
+function saveData(list: Customer[]) {
```

Comment:
Let's try to use arrow functions when possible! To keep consistency
```suggestion
const saveData = (list: Customer[]) => {
```